synapse/handlers/_base.py

```diff
@@ -38,10 +38,10 @@ class BaseHandler:
     """
 
     def __init__(self, hs: "HomeServer"):
-        self.store = hs.get_datastore()  # type: synapse.storage.DataStore
+        self.store: synapse.storage.DataStore = hs.get_datastore()
```

Comment:
Are...these even necessary? Now that `hs` has a type hint I think we should get these automatically? (Same for `state_handler` below).

Reply:
Just looked at HomeServer, yeah, this is probably not necessary as mypy can infer this directly from the `get_datastore` call signature, do you want me to remove it?

Reply:
If it isn't necessary than yep! It is probably from before we had a `hs: "HomeServer"` type hint or something?

Reply:
Probably, i'll have to update the "how to update instance variable type annotations" guidelines on #8351 somewhat to reflect this, but i'll remove this :+1: